### PR TITLE
Add a unit test framework

### PR DESCRIPTION
**Pull request by nachtm**
Status of this:
 - [x] Get a couple of people saying this is ok
 - [x] Move the actual files into the structure
 - [x] Remove DNM title 
 - [x] (hopefully) get some green check marks

This is a first pass at an overall structure with unit tests to this project (see #8). I'd appreciate a couple people trying this out on their machines to make sure that everything makes sense.

I haven't moved any of the actual files MQM into this structure yet -- if this looks good, I can move the files either as part of this PR or a future one. The new structure looks like this:
```
osmquality
    - src
        - mqm
            - source code lives here
    - tests
        - mqm
            - test code lives here, in a package structure that matches source code
    - setup.py
    - tox.ini
    - README.md
    - Pipfile
    - Pipfile.lock
```

A couple of nifty things that we get from this framework:
 - `tox` automatically builds the package for us, so we can be sure that a fresh install of this package works. This means we can catch a lot of packaging errors right off the bat.
 - `tox` also lets us test different versions of python. Right now I have this just set to run 3.7, but we could conceivably expand this in the future.
 - `pytest` uses built-in assert statements, and there's no need to subclass anything -- just make sure that your unit tests start with the word `test`. It also feels very pythonic, which is a big plus in my book.

One downside is that the [import mqm](https://github.com/nachtm/osmquality/blob/4376bef36dc034e3f12cd8d17e7dc63ce42b1af4/tests/mqm/test_sample.py#L1) line will get a red underline in PyCharm (and possibly other IDEs), since tox actually runs the tests in its own virtual environment with mqm installed as a module. PyCharm integrates well with tox, though, so I imagine there's a workaround. I'm in the process of investigating this.

### CODE REVIEW

**Comment by Bentleysb** [comment]
This looks pretty sweet. I tried it out and it worked great. I love how strait forward the assert statements are.

**Comment by danielduhh** [comment]
Do you mind updating the docs; now that we're using a Pipfile

**Comment by nachtm** [comment]
Ok, thanks for the reviews everyone! I'm going to start moving the code into this structure. 

If anyone has time to test out the `pipenv` instructions from the `README.md`, I'd greatly appreciate it! I've tested from as fresh of a copy of this repo as is possible on my computer, but at the end of the day it is just one machine

**Comment by danielduhh** [comment]
Getting an error when running unit tests
```shell
ERROR:  py37: InterpreterNotFound: python3.7
```
Does this require python 3.7 ?

**Comment by danielduhh** [comment]
Also, there's still README language that references the old `requirements.txt` file

File: src/mqm/__init__.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from .sample import *
```
Comment (danielduhh) [inline src/mqm/__init__.py]: This can be removed

File: src/mqm/__init__.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+from .geo_process import *
+from .kd_tree import *
+from .mqm_tool import *
+from .sample import *
```
Comment (danielduhh) [inline src/mqm/__init__.py]: This too

**Comment by nachtm** [comment]
@danielduhh Yes, tox requires you to have an interpreter it can run for each version of python that it tests. I arbitrarily decided to test against 3.7, but I'm very much up for changing that if we'd rather work with 3.6, or something else.

**Comment by savannahostrowski** [comment]
@nachtm I'm not sure if changing the version will make a difference necessarily. Even if we switch to 3.6.x, some folks could run into these same errors (maybe they'll only have 3.7.x installed on their machine). However, the virtual environment will ensure that we keep dependencies clear and stable.

**Comment by danielduhh** [comment]
Yeah, we can stick with 3.7

**Comment by danielduhh** [comment]
@nachtm Do you mind updating the name in `setup.py`

**Pull request merged**